### PR TITLE
Presentation: instance filter dialog

### PR DESCRIPTION
**Pull request by MartynasStrazdas**
Closes https://github.com/iTwin/itwinjs-backlog/issues/498
Updated itwinui-react to 1.46.0, to be able to use `Dialog` component.

### CODE REVIEW

File: presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    const spy = sinon.spy();
+    const { container, getByText, getByDisplayValue } = render(<PresentationInstanceFilterDialog
+      imodel={imodelMock.object}
+      descriptor={descriptor}
+      onInstanceFilterClosed={() => { }}
+      onInstanceFilterApplied={spy}
+      isOpen={true} />);
+
+    const applyButton = container.querySelector<HTMLInputElement>(".presentation-instance-filter-button-bar .iui-high-visibility:disabled");
+    expect(applyButton).to.not.be.null;
```
Comment (saskliutas) [inline presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx:99]: This looke like unnecessary. Check below will faile if this is `null`. Same in other test

File: presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    // select operator
+    const operatorSelector = container.querySelector<HTMLInputElement>(".rule-operator .iui-select-button");
+    expect(operatorSelector).to.not.be.null;
+    fireEvent.click(operatorSelector!);
+    fireEvent.click(getByText(getPropertyFilterOperatorLabel(PropertyFilterRuleOperator.IsNotNull)));
+    // wait until operator is selected
+    await waitFor(() => getByText(getPropertyFilterOperatorLabel(PropertyFilterRuleOperator.IsNotNull)));
+
+    expect(applyButton?.disabled).to.be.false;
+    if (applyButton)
```
Comment (saskliutas) [inline presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx:118]: Unnecessary if. Same in other test

File: presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  });
+  const initialFilter: PresentationInstanceFilterInfo = {
+    filter: {
+      operator: PropertyFilterRuleGroupOperator.And,
+      conditions: [{
+        field: propertiesField,
+        operator: PropertyFilterRuleOperator.IsNull,
+        value: undefined,
+      }],
+    },
```
Comment (saskliutas) [inline presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx:48]:
```suggestion
    filter: {
      field: propertiesField,
      operator: PropertyFilterRuleOperator.IsNull,
      value: undefined,
    },
```

File: presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+      imodel={imodelMock.object}
+      descriptor={descriptor}
+      onInstanceFilterClosed={() => { }}
+      title={<div>{title}</div>}
+      onInstanceFilterApplied={spy}
+      isOpen={true}
+      initialFilter={initialFilter} />);
+
+    const actualTitle = getByText(title);
+    expect(actualTitle).to.not.be.null;
```
Comment (saskliutas) [inline presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx:163]:
```suggestion
    expect(queryByText(title)).to.not.be.null;
```

File: presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+      imodel={imodelMock.object}
+      descriptor={descriptor}
+      onInstanceFilterClosed={() => { }}
+      filterResultCountRenderer={() => { return <div>{count}</div>; }}
+      onInstanceFilterApplied={spy}
+      isOpen={true}
+      initialFilter={initialFilter} />);
+
+    const actualTitle = getByText(count);
+    expect(actualTitle).to.not.be.null;
```
Comment (saskliutas) [inline presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx:181]:
```suggestion
    expect(queryByText(count)).to.not.be.null;
```

**Review by saskliutas — CHANGES_REQUESTED** [review]
Don't forget `extract-api` and `change`

File: presentation/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilterDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+/** @packageDocumentation
+ * @module InstancesFilter
+ */
+import * as React from "react";
+import { Button, Dialog } from "@itwin/itwinui-react";
+import { PresentationInstanceFilterBuilder, PresentationInstanceFilterBuilderProps, PresentationInstanceFilterInfo } from "./PresentationInstanceFilterBuilder";
+import "./PresentationInstanceFilterDialog.scss";
+import { translate } from "../common/Utils";
+
+interface PresentationInstanceFilterDialogProps extends Omit<PresentationInstanceFilterBuilderProps, "onInstanceFilterChanged"> {
```
Comment (saskliutas) [inline presentation/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilterDialog.tsx:14]: Should be exported.

File: presentation/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilterDialog.scss [hunk 7]
```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+@import "~@itwin/itwinui-css/scss/style/theme";
+
+.presentation-instance-filter-dialog {
+
```
Comment (saskliutas) [inline presentation/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilterDialog.scss:8]: Remove empty line

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+      imodel={imodelMock.object}
+      descriptor={descriptor}
+      onClose={() => { }}
+      title={<div>{title}</div>}
+      onApply={spy}
+      isOpen={true}
+      initialFilter={initialFilter} />);
+
+    expect(queryByText(title)).to.not.be.null;
+
```
Comment (saskliutas) [inline presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx:156]: Looks like there are some unnecessary empty lines left in this file. Please remove them

File: presentation/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilterDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+  const onInstanceFilterChanged = React.useCallback((filterInfo?: PresentationInstanceFilterInfo) => {
+    setFilter(filterInfo);
+  }, []);
+
+  const applyButtonHandle = () => {
+    onInstanceFilterApplied(filter);
+  };
+
+  return <Dialog isOpen={isOpen}
```
[the remark this answers is not in the document]
Comment (saskliutas) [reply presentation/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilterDialog.tsx]: There are still some fomatting issues left

**Review by pmconne — APPROVED** [review]

**Review by NancyMcCallB — APPROVED** [review]

**Review by saskliutas — APPROVED** [review]

**Comment by saskliutas** [comment]
@MarcBedard8 @mdastous-bentley can you take a look at changes in `map-layers`?

**Review by mdastous-bentley — APPROVED** [review]

**Pull request merged**